api/opencl_runtime_layer.asciidoc

```diff
@@ -383,8 +383,10 @@ include::{generated}/api/version-notes/CL_QUEUE_PROPERTIES_ARRAY.asciidoc[]
 
 include::{generated}/api/version-notes/CL_QUEUE_SIZE.asciidoc[]
   | cl_uint
-      | Return the currently specified size for the device command-queue.
-        This query is only supported for device command queues.
+      | Return the size of the device command-queue.
+        To be considered valid for this query, _command_queue_ must be a
+        device command-queue.
```

Comment:
I think we should add an entry for the error case below so we keep them all in the same place. Suggest something like:

>{CL_INVALID_COMMAND_QUEUE} if _param_name_ is {CL_QUEUE_SIZE} and  _command_queue_ is not a device command-queue.

Reply:
I thought about this, but I'm wary of replicating information in the error descriptions, and especially listing a bunch of specific enums that could easily grow out-of-date.  Instead, I thought it was better to describe that the only valid command queues for this query are device command queues, and then the invalid command queue error falls out naturally for host command queues.

If that's a little too clever, how would you feel about a more general error condition, something like:

> `CL_INVALID_COMMAND_QUEUE` if _command_queue_ is invalid for _param_name_.

Thanks!

Reply:
I understand the logic but I think it was too indirect. Agree that we should avoid duplication as much as possible but at the same time it is very useful to have all the error cases in one place when writing negative tests or layers/tools. I like your proposal as it achieves both: the list of error conditions is exhaustive _and_ we don't duplicate the details.

Reply:
Sounds good, I've updated the list of error conditions also.  The error code description for `CL_INVALID_COMMAND_QUEUE` is now:

>   * `CL_INVALID_COMMAND_QUEUE` if _command_queue_ is not a valid command-queue, or if _command_queue_ is not a valid command-queue for _param_name_.

Reply:
I'd prefer seeing the new case as a separate bullet point but I'm fine with what you've done.